sdk/servicebus/Azure.Messaging.ServiceBus/src/Core/TransportConnectionScope.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Messaging.ServiceBus.Core
+{
+    internal abstract class TransportConnectionScope : IDisposable
```

Comment:
The reason I added this was to avoid leaking the AMQP specific AMQPConnectionScope in our RetryPolicy. RetryPolicy.RunOperation needs access to the scope to check whether it is Disposed.

Reply:
Ah.... fair enough.   I missed that detail.   Yeah,  the only other idea that I can think of would be to pass a callback to check for disposal, and I like your approach much better.